src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -340,9 +340,50 @@ public static void LogTableStorageDependency(this ILogger logger, string account
             Guard.NotNullOrWhitespace(tableName, nameof(tableName));
             Guard.NotNullOrWhitespace(accountName, nameof(accountName));
 
+            context = context ?? new Dictionary<string, object>();
+
             logger.LogInformation(DependencyFormat, "Azure table", tableName, accountName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
         }
 
+        /// <summary>
+        ///     Logs an Azure Event Hub Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="accountName">Account of the resource</param>
+        /// <param name="eventHubName">Name of the Event Hub resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="measurement">Measuring the latency to call the dependency</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogEventHubsDependency(this ILogger logger, string accountName, string eventHubName, bool isSuccessful, DependencyMeasurement measurement, Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger));
+            Guard.NotNullOrWhitespace(accountName, nameof(accountName));
+            Guard.NotNullOrWhitespace(eventHubName, nameof(eventHubName));
+
+            LogEventHubsDependency(logger, accountName, eventHubName, isSuccessful, measurement.StartTime, measurement.Elapsed, context);
+        }
+
+        /// <summary>
+        ///     Logs an Azure Event Hub Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="accountName">Account of the resource</param>
+        /// <param name="eventHubName">Name of the Event Hub resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="startTime">Point in time when the interaction with the dependency was started</param>
+        /// <param name="duration">Duration of the operation</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogEventHubsDependency(this ILogger logger, string accountName, string eventHubName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
```

Comment:
Event Hubs uses the concept of namespaces similar to Service Bus, so we should name it accordingly.
```suggestion
        public static void LogEventHubsDependency(this ILogger logger, string namespaceName, string eventHubName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
```

Reply:
And can I use the `namespaceName` as dependency data?

Reply:
Updated.

Reply:
Sure!